cluster/cluster.go

```diff
@@ -39,54 +39,106 @@ type Cluster struct {
 	ContentIDs []int
 	Hostnames  []string
 	Segments   []SegConfig
-	ByContent  map[int]*SegConfig
+	ByContent  map[int][]*SegConfig
 	ByHost     map[string][]*SegConfig
 	Executor
 }
 
 type SegConfig struct {
 	DbID      int
 	ContentID int
+	Role      string
 	Port      int
 	Hostname  string
 	DataDir   string
 }
 
 /*
- * We pass values from this enum into ShellCommands, RemoteOutputs, and associated
- * functions to define the nature and scope of the command execution.
- * - ON_SEGMENTS:            Execute on each segment, excluding the master.
- * - ON_SEGMENTS_AND_MASTER: Execute on each segment, including the master.
- * - ON_HOSTS:               Execute on each host, excluding the master host.
- * - ON_HOSTS_AND_MASTER:    Execute on each host, including the master host.
+ * A "scope" is a value composed of one or more of the below constants that is
+ * passed into ShellCommands, RemoteOutputs, and related structs and functions
+ * to define the scope of the command execution.  The meaning of each value is
+ * as follows:
  *
- * - ON_MASTER_TO_SEGMENTS:            Execute commands on master about segments, excluding master.
- * - ON_MASTER_TO_SEGMENTS_AND_MASTER: Execute commands on master about segments, including master.
- * - ON_MASTER_TO_HOSTS:               Execute commands on master about hosts, excluding master.
- * - ON_MASTER_TO_HOSTS_AND_MASTER:    Execute commands on master about hosts, including master.
+ * ON_SEGMENTS:     Execute one command per segment.
+ * ON_HOSTS:        Execute one command per host.
+ *
+ * INCLUDE_MASTER:  Include the master host or segment in the command list.
+ * EXCLUDE_MASTER:  Exclude the master host or segment from the command list.
+ *
+ * ON_REMOTE:       Execute each command on the specified remote segment/host.
+ * ON_LOCAL:        Execute all commands on the master host.
```

Comment:
One final note about documentation: the comment at the top of the file should probably point folks straight to `GenerateAndExecuteCommand()` because with most of the functions being public, it can be confusing trying to figure out where to start using this library.